polly/plugins/maya/publish/submit_deadline.py

```diff
@@ -25,7 +25,8 @@ def process(self, instance):
         from avalon import api
         from avalon.vendor import requests
 
-        assert api.Session["AVALON_DEADLINE"], "Requires AVALON_DEADLINE"
+        deadline = api.Session.get("AVALON_DEADLINE", None)
```

Comment:
Good call.

This remained from before `api.Session` only included available keys, before it included DEADLINE even if it wasn't set.

One small critique, could we make the variable name `AVALON_DEADLINE`? I found that it makes for more clear reading and highlights that it's coming from the environment/session.